internal/cmd/platform/register_platform.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package platform
+
+import (
+	"fmt"
+
+	"github.com/Peripli/service-manager-cli/internal/cmd"
+	"github.com/Peripli/service-manager-cli/internal/output"
+	"github.com/Peripli/service-manager-cli/pkg/types"
+
+	"github.com/Peripli/service-manager-cli/pkg/smclient"
+
+	"github.com/spf13/cobra"
+)
+
+// RegisterPlatformCmd wraps the smctl register-platform command
+type RegisterPlatformCmd struct {
+	*cmd.Context
+
+	platform types.Platform
+
+	outputFormat int
+}
+
+// NewRegisterPlatformCmd returns new register-platform command with context
+func NewRegisterPlatformCmd(context *cmd.Context) *RegisterPlatformCmd {
+	return &RegisterPlatformCmd{Context: context, platform: types.Platform{}}
+}
+
+// SetSMClient set SM client
+func (rpc *RegisterPlatformCmd) SetSMClient(client smclient.Client) {
+	rpc.Client = client
+}
+
+// SetOutputFormat set command's output format
+func (rpc *RegisterPlatformCmd) SetOutputFormat(format int) {
+	rpc.outputFormat = format
+}
+
+// HideUsage hide command's usage
+func (rpc *RegisterPlatformCmd) HideUsage() bool {
+	return true
+}
+
+// Command returns cobra command
+func (rpc *RegisterPlatformCmd) Command() *cobra.Command {
+	result := rpc.buildCommand()
+	result = rpc.addFlags(result)
+
+	return result
+}
+
+func (rpc *RegisterPlatformCmd) buildCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:     "register-platform [name] [type] <description>",
+		Aliases: []string{"rp"},
+		Short:   "Registers a platform",
+		Long:    `Registers a platform`,
+
+		PreRunE: cmd.PreRunE(rpc, rpc.Context),
+		RunE:    cmd.RunE(rpc),
+	}
+}
+
+func (rpc *RegisterPlatformCmd) addFlags(command *cobra.Command) *cobra.Command {
+	command.Flags().StringVarP(&rpc.platform.ID, "id", "i", "", "external platform ID")
```

Comment:
Not for the cli but allowing a platform to be created by specifying the id requires a validation. I can execute `smctl register-platform zxc zxcsdf -i aaa_(!@#` and I can no longer retrieve this platform because `/v1/platforms/aaa_(!@#` returns 404 which is a problem because now I can't delete it because `smctl delete-platform zxc` executes `DELETE /v1/platforms/aaa_(!@#` which returns 404